### PR TITLE
Documentation: update README

### LINKED ISSUE #142: Add NLopt and BAT to README
README.md should contain short sections how to install and prepare NLopt and BAT. As both components are optional this is not super critical right now.

### PR DESCRIPTION
**Pull request by suhlatwork**
Add a section explaining the installation of NLopt and BAT.

Closes #142 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +165,46 @@ The build system tries to find your Python installation automatically. For this
...
+### BAT (optional) ###
+
+An efficient way to create Monte Carlo events according to a given model can be used if the _Bayesian Analysis Toolkit_ (<https://github.com/bat/bat>) is available. The ROOTPWA build system is able to automatically detect BAT if it is either installed in system paths or if the `BATINSTALLDIR` environment variable is set.
+
+1.  At the time of writing this documentation there is no tagged release of BAT working in our usecase. The current `git` `HEAD` should be used instead.
+
+    `> git clone https://github.com/bat/bat.git`
+
+2.  Change to the directory that was created during the clone to configure and compile BAT.
+
```
Comment (bgrube) [inline README.md]: Please add
`> ./autogen.sh`

File: README.md [hunk 2]
```diff
@@ -165,6 +165,46 @@ The build system tries to find your Python installation automatically. For this
 In addition you need to compile the `Boost.Python` library (e.g. by running the supplied `compileBoostLibraries.sh` script; see "Compiling Boost Library" above). Make also sure that the ROOT installation you are using was compiled with Python support (running `root-config --features` should list `python`) against the _same_ Python version you are using (`ldd ${ROOTSYS}/lib/libPyROOT.so | grep -i python` shows you the Python library version against which ROOT was linked). Make also sure that your `PYTHONPATH` environment variable includes `${ROOTSYS}/lib`.
 
 
+### NLopt (optional) ###
+
+The _NLopt_ library (<http://ab-initio.mit.edu/wiki/index.php/NLopt>) provides a faster minimizer compared to the default Minuit2. The ROOTPWA build system is able to automatically detect and use the library if it is either installed in a system directory or if the `NLOPT` environment variable is defined.
+
+1.  Download the source tarball from <http://ab-initio.mit.edu/wiki/index.php/NLopt> and extract it to a directory of your choice.
+
+2.  Configure and compile _NLopt_.
```
Comment (bgrube) [inline README.md]: Please change:
Configure, compile, and install

File: README.md [hunk 3]
```diff
@@ -165,6 +165,46 @@ The build system tries to find your Python installation automatically. For this
...
+
+    `> ./configure --with-cxx --enable-shared && make`
```
Comment (bgrube) [inline README.md]: Please add
`--prefix=/your/folder/to/install/nlopt`
`&& make install`

File: README.md [hunk 4]
```diff
@@ -165,6 +165,46 @@ The build system tries to find your Python installation automatically. For this
 In addition you need to compile the `Boost.Python` library (e.g. by running the supplied `compileBoostLibraries.sh` script; see "Compiling Boost Library" above). Make also sure that the ROOT installation you are using was compiled with Python support (running `root-config --features` should list `python`) against the _same_ Python version you are using (`ldd ${ROOTSYS}/lib/libPyROOT.so | grep -i python` shows you the Python library version against which ROOT was linked). Make also sure that your `PYTHONPATH` environment variable includes `${ROOTSYS}/lib`.
 
 
+### NLopt (optional) ###
+
+The _NLopt_ library (<http://ab-initio.mit.edu/wiki/index.php/NLopt>) provides a faster minimizer compared to the default Minuit2. The ROOTPWA build system is able to automatically detect and use the library if it is either installed in a system directory or if the `NLOPT` environment variable is defined.
```
Comment (bgrube) [inline README.md]: Maybe one can add a sentence like
The easiest way is to install the library into the same directory as the source.

File: README.md [hunk 5]
```diff
@@ -165,6 +165,46 @@ The build system tries to find your Python installation automatically. For this
...
+
+An efficient way to create Monte Carlo events according to a given model can be used if the _Bayesian Analysis Toolkit_ (<https://github.com/bat/bat>) is available. The ROOTPWA build system is able to automatically detect BAT if it is either installed in system paths or if the `BATINSTALLDIR` environment variable is set.
+
+1.  At the time of writing this documentation there is no tagged release of BAT working in our usecase. The current `git` `HEAD` should be used instead.
+
+    `> git clone https://github.com/bat/bat.git`
+
+2.  Change to the directory that was created during the clone to configure and compile BAT.
+
+    `> ./configure && make`
```
Comment (bgrube) [inline README.md]: same as for NLopt: prefix and make install

**Comment by bgrube** [comment]
While you are at it: It would be nice if you could add a sentence similar to the one below to the "Contributing Code" section at the end

The development work of ROOTPWA is organized via [Trello](https://trello.com/b/MdWlJZPQ).

**Comment by suhlatwork** [comment]
Updated the README following your requests.

**Comment by suhlatwork** [comment]
Update the README again.

**Comment by bgrube** [comment]
Merged, thank you.

**Pull request merged**